library/Icinga/Application/Hook/Common/DbMigration.php

```diff
@@ -0,0 +1,151 @@
+<?php
+
+/* Icinga Web 2 | (c) 2023 Icinga GmbH | GPLv2+ */
+
+namespace Icinga\Application\Hook\Common;
+
+use ipl\Sql\Connection;
+use RuntimeException;
+
+class DbMigration
```

Comment:
I would rename that to `DbMigrationScript` or `DbMigrationStep`. Sure, it's in a different namespace, but since you're also implementing the hook as `DbMigration`, I don't think the big differences between those classes are reflected by the namespace.